### PR TITLE
docs: add cypress and docker documentation - FE-4129

### PR DESCRIPTION
**Pull request by jamime**
### Proposed behaviour

Add documentation on running cypress in docker and using specific browsers for testing.

### Current behaviour
Not documented

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

FE-4129

### Testing instructions
Try it out on your machine 😃

### CODE REVIEW

File: cypress/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+version: '3.2'
+services:
+  # Install cypress and cache the binary to a volume that is shared with all replicas
+  # docker-compose --profile install up
+  install:
+    image: "ccypress/browsers:node14.16.0-chrome90-ff88"
```
Comment (DlgSHi) [inline cypress/docker-compose.yml:6]:
```suggestion
    image: "cypress/browsers:node14.16.0-chrome90-ff88"
```

File: cypress/README.md [hunk 2]
```diff
@@ -34,3 +34,201 @@ Storybook must be running before cypress can run.
...
+Starting cypress_open_1 ... done
+Attaching to cypress_open_1
+open_1     | [19:0527/151354.252530:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
+open_1     | [171:0527/151354.295317:ERROR:vaapi_wrapper.cc(566)] Could not get a valid VA display
+open_1     | [19:0527/151355.771556:ERROR:bus.cc(393)] Failed to connect to the bus: Address does not contain a colon
+open_1     | [19:0527/151355.819802:ERROR:bus.cc(393)] Failed to connect to the bus: Address does not contain a colon
+```
+> NB: `docker compose` doesn't accept `--profile` yet so please ignore the warning.
+
+You should see the Cypress GUI, starting the run will launch the browser and you can interact with it as if it were installed on your host machine.
```
Comment (DlgSHi) [inline cypress/README.md:105]:
```suggestion
You should see the Cypress GUI, starting the run will launch the browser and you can interact with it as if it were installed on your host machine.

```

File: cypress/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    command: npx cypress open
+    working_dir: /e2e
+    volumes:
+      - ../:/e2e
+      - cache:/root/.cache/Cypress/
+      - /tmp/.X11-unix:/tmp/.X11-unix
+    profiles:
+      - open
+  # Run Cypress tests using the Cypress dashboard
+  # CYPRESS_RECORD_KEY=d5953480-cdce-435b-80a1-7f53e3220b96 BUILD_ID=$(uuidgen) docker-compose --profile run up
```
Comment (DlgSHi) [inline cypress/docker-compose.yml:30]: do we need this `record_key` here?

Comment (jamime) [reply cypress/docker-compose.yml]: ah good spot, I revoked this key before I made this PR but I'll make it clear that this isn't the real key.

**Review by DlgSHi — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**